apps/sms/js/compose.js

```diff
@@ -320,13 +323,18 @@ var Compose = (function() {
     attachments.delete(attachmentNode);
   }
 
+  function replaceAttachmentNode(newNode, oldNode) {
```

Comment:
Wondering why do we need separate file-level function for this code that is used only once anyway?

What if we add ```attachmentNode.remove()``` to ```disposeAttachmentNode```, so as the result we'll have some local functions:
```
function onRemove() {
  disposeAttachmentNode(this.currentAttachmentDOM);

  state.size = null;
  onContentChanged();
}

function onReplace() {
  this.requestAttachment().then((newAttachment) => {
    dom.message.insertBefore(
      insert(newAttachment),
      this.currentAttachmentDOM
    );

    disposeAttachmentNode(this.currentAttachmentDOM);

   onContentChanged(newAttachment);
  }, this._onAttachmentRequestError);
}

// and then
options.items = [
  { l10nId: `view-attachment-${fileType}`, method: onView },
  { l10nId: `remove-attachment-${fileType}`, method: onRemove },
  { l10nId: `replace-attachment-${fileType}`, method: onReplace },
  { l10nId: 'cancel' }
]

```

Anyway I'm fine with what you have now, so it's up to you :)